### PR TITLE
implement clone for y documents

### PR DESCRIPTION
**Pull request by filipot**
Dunno if there are some edge cases that I don't know about, but it seems to pass the tests

### CODE REVIEW

File: src/utils/Doc.js [hunk 1]
```diff
@@ -227,6 +227,23 @@ export class Doc extends Observable {
...
+
+  /**
+   * Clone all shared data types into a fresh Y.Doc instance.
+   *
+   * @return {Doc}
+   */
+  clone({ guid = random.uuidv4(), collectionid = null, gc = true, gcFilter = () => true, meta = null, autoLoad = false, shouldLoad = true } = {}){
+    const cloned = new Doc({guid, collectionid, gc, gcFilter, meta, autoLoad, shouldLoad});
+    this.share.forEach((value, key) => {
+        const clonedType = value.clone();
```
[the remark this answers is not in the document]
Comment (filipot) [reply src/utils/Doc.js:239]: Doesn't `clonedType._integrate(this, null);` fix this? It seems to work fine: [Changing values of cloned yDoc repl](https://svelte.dev/repl/b911b9be436b41589bb0d81707e4dd91?version=3.44.2)

The main reason I want this clone function is to get the leanest version of the document, without any tombstones or history (which from my understanding `Y.applyUpdate(cloned, Y.encodeStateAsUpdate(ydoc))` keeps)

**Comment by toantd90** [comment]
@dmonad It is very helpful feature. How can we achieve it with the latest yjs?
